src/index.tsx

```diff
@@ -2155,6 +2168,44 @@ class App extends React.Component<{}, AppState> {
             skipHistory = true;
             this.forceUpdate();
           }}
+          onDoubleClick={e => {
+            const x =
+              e.clientX - CANVAS_WINDOW_OFFSET_LEFT - this.state.scrollX;
+            const y = e.clientY - CANVAS_WINDOW_OFFSET_TOP - this.state.scrollY;
+
+            if (getElementAtPosition(x, y)) {
+              return;
+            }
+
+            const element = newElement(
+              "text",
+              x,
+              y,
+              this.state.currentItemStrokeColor,
+              this.state.currentItemBackgroundColor,
+              "hachure",
+              1,
+              1,
+              100
+            );
+
+            if (isTextElement(element)) {
```

Comment:
You don’t need this, you literally created a text element on the statement before :)

Reply:
Okay, I did this because the cast to TextElement, but I'll cast directly